server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -668,6 +668,14 @@ public boolean isES87TSDBCodecEnabled() {
         Property.Final
     );
 
+    public static final Setting<Boolean> LOGSDB_SORT_ON_MESSAGE_TEMPLATE = Setting.boolSetting(
+        "index.logsdb.sort_on_message_template",
+        false,
```

Comment:
I think disabling by default is the right choice, given that we don't fully know the impact of this change index sorting on replacing points wiht doc value skippers on `@timestamp`.